include/stdx/tuple_algorithms.hpp

```diff
@@ -16,6 +16,29 @@
 
 namespace stdx {
 inline namespace v1 {
+template <typename F, tuplelike... Ts> constexpr auto apply(F &&f, Ts &&...ts) {
```

Comment:
question: would it be worthwhile to write this in terms of `tuple_cat` or vis-a-versa?

Reply:
And the reason I didn't DRY `element_indices` is that I didn't want to adversely affect the compile-time performance of `tuple_cat`. It's a workhorse for CIB. Where variadic `apply` is something that I currently have one use case for (`when_all`).